### ISSUE TITLE
WebRTC SRTP performance?

### ISSUE THREAD

**Issue by seeme-o** [WolverinDEV/DataPipes#1]
I'm curious about the performance and resource requirements of SRTP streams using this lib. Have you done much testing or even just general experiences you can share? Thanks

**Comment by WolverinDEV** [comment]
Hey,
lovely to see your interest within this library.

As you've may already noticed, this library is currently more a utility for various protocols which I need for  a lot of my projects. And the need for a simple library which support these protocols is also the motivative behind this project.

The WebRTC part of the library (currently `datachannel` and `audio` support only) is manly designed for being used within a server software. So the focus is on performance and and minimum resource usage.
A key point here was to build up a light wight API which implement the WebRTC protocol standard without having a need for the official C++ API (https://webrtc.org/native-code/) which features a lot more then you actually need for an server (like audio & video interfaces).

Out of my personal experience I could say that DataPipes usies less resources then the official C++ API specially when it comes to simultaneously sessions. But this is just a relative feeling, I haven't yet made any benchmarks.

So in conclusion, feel free to explore and adopt parts and may even contribute to this project.
Various examples how you could use this library could be found here:
https://github.com/WolverinDEV/DataPipes/tree/master/test (Yes its called test... I know I should change this)

**Comment by ClosetGeek-Git** [comment]
I've been surfing through this project for a week now and it's looking promising. Keep up the good work!

One thing, your lacking a license file.

**Comment by WolverinDEV** [comment]
Its lovely to hear :)
If you have any question feel free to ask here, or on my TeaSpeak server ts.teaspeak.de :)

**Comment by WolverinDEV** [comment]
I'm closing the issue now since I've dropped the whole WebRTC part in this.
The project itself was more like trail and error and I'd never fully thought though the API.

I've recoded everything in [Rust](https://en.wikipedia.org/wiki/Rust_(programming_language)) which turns out to be way better in nearly any terms like performance, API design, functionality (now supporting video, negotiations support for nack etc)
